test/e2e/framework/framework.go

```diff
@@ -296,19 +361,22 @@ func (f *Framework) afterEach() {
 			return
 		}
 
-		f.deleteNamespace(ctx, f.namespace)
+		f.deletePrimaryNamespace(ctx)
```

Comment:
wouldn't it be cleaner to delete every secondary namespace first, and then primary? Here you're letting GC to clean up, which may end up being cleared after test finishes. If something is off, we would succeed the test, but secondary ns would be dangling.

Reply:
Are you suggesting removing the owner references altogether or just adding the explicit deletion on top of that?

Reply:
Added explicit deletion

Reply:
owner references are good, we need those. I suggest explicit remove, to clean everything we create before we call it success. Relying on GC is good when test it interrupted.